dibi/drivers/DibiOracleDriver.php

```diff
@@ -365,11 +365,12 @@ public function getResultColumns()
 		$count = oci_num_fields($this->resultSet);
 		$columns = array();
 		for ($i = 1; $i <= $count; $i++) {
+			$type = oci_field_type($this->resultSet, $i);
 			$columns[] = array(
 				'name' => oci_field_name($this->resultSet, $i),
 				'table' => NULL,
 				'fullname' => oci_field_name($this->resultSet, $i),
-				'nativetype'=> oci_field_type($this->resultSet, $i),
+				'nativetype' => $type === 'NUMBER' && oci_field_scale($this->resultSet, $i) === 0 ? 'INTEGER' : $type,
```

Comment:
@dg I'm not sure if `oci_field_type()` can ever return `INTEGER` string. I always got `NUMBER(x, 0)` for any kind of integer, so this is hacking a little bit.